models/offer.go

```diff
@@ -56,35 +91,42 @@ func GetOfferByID(db runner.Connection, gameID, id string, mr *MixedMetricsRepor
 	return &offer, err
 }
 
-//GetEnabledOffers returns all the enabled offers
-func GetEnabledOffers(db runner.Connection, gameID string, offersCache *cache.Cache, expireDuration time.Duration, mr *MixedMetricsReporter) ([]*Offer, error) {
+//GetEnabledOffers returns all the enabled offers and matching offers
+func GetEnabledOffers(db runner.Connection, gameID string, offersCache *cache.Cache, expireDuration time.Duration, filterAttrs map[string]string, mr *MixedMetricsReporter) ([]*Offer, error) {
 	var offers []*Offer
 	var err error
 
 	enabledOffersKey := GetEnabledOffersKey(gameID)
-	offersInterface, found := offersCache.Get(enabledOffersKey)
-
-	if found {
-		//fmt.Println("Offers Cache Hit")
-		offers = offersInterface.([]*Offer)
-		return offers, err
+	// TODO: See if it is possible to enable cache with filters
+	if len(filterAttrs) == 0 {
+		offersInterface, found := offersCache.Get(enabledOffersKey)
+
+		if found {
+			//fmt.Println("Offers Cache Hit")
```

Comment:
Could you add some logs here? We'd like to have 3 types of logs so we can build dashboards:
- should use cache
- cache hit (found)
- cache miss (!found)

Maybe we could add a 4th type indicating non-cacheable requests.